### DISCUSSION TITLE
Low overhead crash-proof logging: a sink for writing to shared memory?

### DISCUSSION THREAD

**Discussion by sketch34** [gabime/spdlog#2962] [Q&A] [answered]
Are there any spdlog sink implementations that write to shared memory?

The spdlog library is great and I have used it on a couple of projects. We are considering using it to replace the logging system in a high-performance server. At the same time, we would like to achieve two things:
1) Minimize the performance impact of logging.
2) If the server process crashes, we must see every log message up to the crash point.

One way to achieve (2) is flushing every log message, but this is slow. An alternative is to do unbuffered logging directly into shared memory that is then read by another process (the log "consumer") in order to display the logs. The consumer process could also handle the formatting work. This approach achieves both (1) and (2).

In case we would reinvent the wheel, we were wondering if there is any existing spdlog sink that does this, or the best starting point. All I found so far was this: https://github.com/gabime/spdlog/discussions/2888.

Also the same idea is sketched out here: https://stackoverflow.com/a/71206989/988870

**Comment by tt4g** [comment]
> Are there any spdlog sink implementations that write to shared memory?
Not supported.

> In case we would reinvent the wheel, we were wondering if there is any existing spdlog sink that does this, or the best starting point. All I found so far was this: https://github.com/gabime/spdlog/discussions/2888.

You can create a SINK that fits your purpose.
Wiki: https://github.com/gabime/spdlog/wiki/4.-Sinks#implementing-your-own-sink

**Comment by gabime** [comment, marked as answer]
This seems good for this: https://github.com/gabime/spdlog/pull/2604
